lib/semian/mysql2.rb

```diff
@@ -57,6 +61,17 @@ def semian_identifier
       end
     end
 
+    def ping
+      result = nil
+      acquire_semian_resource(adapter: :mysql, scope: :ping) do
+        result = raw_ping
+        raise PingFailure.new(result.to_s) unless result
+      end
+      result
+    rescue ResourceBusyError, CircuitOpenError, PingFailure
+      false
```

Comment:
Ok, so I could probably answer this myself if I checked (which I might do a little later), but I think there is a possible important concern here.

If `ping` is used by connection the pool to check if a connection is healthy, and create a new one if it isn't. We should make sure failing pings when the circuit is open won't make AR's connection pool do crazy things, and generate tons of connections attempts. (I'm pretty sure the connection is also protected by the circuit breaker, but I prefer to be on the safe side)